### ISSUE TITLE
Using grad with BatchNorm fails at the second iteration

### ISSUE THREAD

**Issue by JoaoMarcosCSilva** [patrick-kidger/equinox#183]
Hi there!

It seems that BatchNorm doesn't work inside a gradient transformation after the first time it's called. For instance, see the following code:

```python
import equinox as eqx
from jax import numpy as jnp
import jax

net = eqx.experimental.BatchNorm(input_size=10, axis_name="batch")


def loss_fn(net, x):
    return jax.vmap(net, axis_name="batch")(x).mean()


grad_fn = eqx.filter_grad(loss_fn)

# This works as expected
grad_fn(net, jnp.ones((32, 10)))

# This throws "NotImplementedError: JVP rule is implemented only for id_tap, not for call"
grad_fn(net, jnp.ones((32, 10)))
```

The second call to `grad_fn` above throws a `NotImplementedError: JVP rule is implemented only for id_tap, not for call` error.

**Comment by patrick-kidger** [comment]
Hey there! Yep, this is a known bug: #166, hopefully fixed in #172.

In the mean time you can try installing Equinox from the #172 branch, and seeing if that fixes your downstream use-case.

**Comment by JoaoMarcosCSilva** [comment]
Thanks! Changing to that branch did solve the issue
